core/src/main/scala/dx/core/languages/cwl/CwlBundle.scala

```diff
@@ -107,7 +107,9 @@ object CwlBundle {
         val newWorkflows = workflows + (wf.name -> wf)
         wf.steps.foldLeft(tools, expressions, newWorkflows, newReqs, newHints, newParents) {
           case ((toolAccu, exprAccu, wfAccu, reqAccu, hintAccu, parentAccu), step) =>
-            val newParent = currentParent ++ CwlDxName.fromDecodedName(step.frag).getDecodedParts
+            val newParent = currentParent ++
```

Comment:
https://github.com/dnanexus/dxCompiler/blob/63964655245445f172e360499e53defcd3f57d55/executorCwl/src/main/scala/dx/executor/cwl/CwlWorkflowExecutor.scala#L527-L531

> the target parameter would be in the format  `{proc}#{path}`, where proc is the top-level process name and path is the path from the top-level process to the target. For example, if the top-level workflow is 'wf' and we want to run a step in a sub-workflow, it might be 'wf#step1/nested/step2'

The previous implementation assumed the `step.frag = "nested/step2"` . However if the nested wf has a name ended with 'cwl', it might fail to add the nested wf name to the new parent vector, because now `step.frag=step2` and nested.cwl is recognized as the new namespace where `step2` is defined.

Therefore